src/lib/spsp.js

```diff
@@ -31,24 +38,100 @@ const _querySPSP = function * (receiver) {
+  assert(typeof response.ledger_info.currency_symbol === 'string', 'ledger_info.currency_symbol must be a string')
+  assert(typeof response.ledger_info.scale === 'number', 'ledger_info.scale must be a number')
+  assert(typeof response.ledger_info.precision === 'number', 'ledger_info.precision must be a number')
+  assert(typeof response.receiver_info === 'object', 'receiver_info must be an object')
+}
+
+const _createPayment = (plugin, spsp, quote, id) => {
+  const sourceAmount =
+    toDecimal(quote.sourceAmount, plugin.getInfo().scale)
+  const destinationAmount =
+    toDecimal(quote.destinationAmount, spsp.ledger_info.scale)
+
+  return {
+    id: id || uuid(),
+    sourceAmount: sourceAmount,
+    destinationAmount: destinationAmount,
+    destinationAccount: spsp.destination_account,
+    connectorAccount: quote.connectorAccount,
+    sourceExpiryDuration: quote.sourceExpiryDuration,
+    spsp: spsp
+  }
 }
 
+/**
+  * Query an SPSP endpoint and get SPSP details
+  *
+  * @param {String} receiver webfinger account identifier (eg. 'alice@example.com') or URL to SPSP endpoint.
+  *
+  * @return {Promise<SpspResponse>} SPSP SPSP response from server
+  */
+
 const query = co.wrap(_querySPSP)
 
-const _quote = function * ({ plugin, spsp, sourceAmount, destinationAmount, ilp }) {
-  if (!plugin) throw new Error('missing plugin')
-  if (!spsp.destination_account) throw new Error('missing destination account')
-  if (!spsp.maximum_destination_amount) throw new Error('missing maximum destination amount')
-  if (!spsp.minimum_destination_amount) throw new Error('missing minimum destination amount')
+/**
+  * Quote to an SPSP receiver
+  *
+  * @param {Object} plugin Ledger plugin used for quoting.
+  * @param {Object} params Quote parameters
+  * @param {String} params.receiver webfinger account identifier (eg. 'alice@example.com') or URL to SPSP endpoint.
+  * @param {String} [params.sourceAmount] source amount to quote. This is a decimal, NOT an integer. It will be shifted by the sending ledger's scale to get the integer amount.
+  * @param {String} [params.destinationAmount] destination amount to quote. This is a decimal, NOT an integer. It will be shifted by the receiving ledger's scale to get the integer amount.
+  * @param {Array} [params.connectors=[]] connectors to quote. These will be supplied by plugin.getInfo if left unspecified.
+  * @param {String} [params.id=uuid()] id to use for payment. sending a payment with the same id twice will be idempotent. If left unspecified, the id will be generated randomly.
+  * @param {Number} [params.timeout=5000] how long to wait for a quote response (ms).
+  *
+  * @returns {Promise<SpspPayment>} SPSP payment object to be sent.
+  */
 
-  const client = new IlpCore.Client(plugin, ilp)
-  const quote = yield client.quote({
+const quote = function * (plugin, {
+  receiver,
+  sourceAmount,
+  destinationAmount,
+  connectors,
+  id,
+  timeout
+}) {
```

Comment:
If you pass in a memo here it won't actually be used so you'd need to first quote the payment params and then add the memo to the response. That doesn't seem great. (There should also be a test to make sure this works properly)